### PR TITLE
Add coverage for job templates functionality

### PR DESCRIPTION
**Pull request by oshtaier**
```
nosetests tests/foreman/ui/test_remoteexecution.py
........SSSSSS..
----------------------------------------------------------------------
Ran 16 tests in 326.166s

OK (SKIP=6)
```

### CODE REVIEW

**Comment by oshtaier** [comment]
Part of #3624

File: robottelo/ui/job_template.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            effective_user_value=effective_user_value,
+            current_user=current_user, overridable=overridable,
+            snippet=snippet
+        )
+        self.click(common_locators['submit'])
+
+    def add_input(self, job_template_name, name, required=None,
+                  input_type='User input', advanced=None, options=None,
+                  description=None):
+        """Add new input to existing job template"""
```
Comment (sghai) [inline robottelo/ui/job_template.py]: Does this function defined to support only `User input` input_type ? Just curious about other input types. In any case if this function going to support only `User input` type then I think we should update the docstrings too. thanks

**Comment by sghai** [comment]
ACK pending a small non-blocking comment and  thanks for implementing CRUD functions and tests around job templates. Looks good to me.

@JacobCallahan Hey. Could you please take a look and review the tests around job templates ?

File: robottelo/ui/job_template.py (hunk 1 above)
Comment (oshtaier) [inline robottelo/ui/job_template.py]: At that moment - yes, but it is extensible. You can add support of any other type, but that will require adding necessary locators

**Comment by oshtaier** [comment]
And, forgot to add that now we support input into ACE editors. That one line improvement actually took most of the time for that PR :)

File: tests/foreman/ui/test_remoteexecution.py [hunk 2]
```diff
@@ -61,12 +117,52 @@ def test_positive_create_job_template_input(self):
...
+                name=name,
+                template_type='input',
+                template_content=gen_string('alphanumeric', 20),
+            )
+            self.assertIsNotNone(self.jobtemplate.search(name))
+            self.jobtemplate.add_input(name, var_name)
+            self.jobtemplate.update(
+                name,
+                template_type='input',
+                template_content='<%= input("{0}") %>'.format(var_name)
```
Comment (JacobCallahan) [inline tests/foreman/ui/test_remoteexecution.py]: this template content should actually look like this ```'<%= input(:{0}) %>'.format(var_name)```
which would result in the template code looking like this
```<%= input(:errata) %>```

File: tests/foreman/ui/test_remoteexecution.py [hunk 3]
```diff
@@ -181,11 +300,31 @@ def test_positive_preview_verify(self):
...
+                name=name,
+                template_type='input',
+                template_content=gen_string('alpha'),
+                org=self.organization.name,
+            )
+            self.jobtemplate.add_input(name, var_name)
+            self.jobtemplate.update(
+                name,
+                template_type='input',
+                template_content='<%= input("{0}") %>'.format(var_name)
```
Comment (JacobCallahan) [inline tests/foreman/ui/test_remoteexecution.py]: same feedback as previously noted for the input

**Comment by JacobCallahan** [comment]
ACK. This looks great! I was going to suggest a different formatting for input controls, but these work just was well.

**Pull request merged**